crates/core/src/config.rs

```diff
@@ -42,62 +38,334 @@ const QUALIFIER: &str = "";
 const ORGANIZATION: &str = "The Freenet Project Inc";
 const APPLICATION: &str = "Freenet";
 
+#[derive(clap::Parser, Debug, Serialize, Deserialize)]
+pub struct ConfigArgs {
+    /// Node operation mode.
+    #[clap(value_enum, env = "MODE")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub mode: Option<OperationMode>,
+
+    /// Path to the configuration file.
+    #[clap(short, long)]
+    #[serde(skip)]
+    pub config: Option<PathBuf>,
+
+    #[clap(flatten)]
+    #[serde(flatten)]
+    pub gateway: GatewayArgs,
+    #[clap(value_parser, env = "TRANSPORT_KEYPAIR")]
+    pub transport_keypair: Option<PathBuf>,
```

Comment:
what we want is to deserialise this into a "TransportKeypair" and if it doesn't exist create one (`::new`). Error out in case is not a proper key.

Should try to deserialise it from a valid pkcs1 file https://docs.rs/pkcs1/0.7.5/pkcs1/index.html